src/ring_theory/localization.lean

```diff
@@ -728,7 +728,23 @@ def non_zero_divisors : submonoid R :=
     have z * x₁ * x₂ = 0, by rwa mul_assoc,
     hx₁ z $ hx₂ (z * x₁) this }
 
-variables {A : Type*} [integral_domain A]
+variables {R}
+
+lemma mul_mem_non_zero_divisors {a b : R} :
+  a * b ∈ non_zero_divisors R ↔ a ∈ non_zero_divisors R ∧ b ∈ non_zero_divisors R :=
```

Comment:
This is just because `non_zero_divisors` is a prime ideal. Do we already have that fact?

Reply:
It's not really an ideal because `0` is a zero divisor, unfortunately. I can't find a definition of `submonoid.is_prime`, should I add this (in another PR?)

Reply:
I didn't realise this was about monoids sorry.